components/script/dom/htmlinputelement.rs

```diff
@@ -354,8 +360,18 @@ impl HTMLInputElementMethods for HTMLInputElement {
                             |a| DOMString::from(a.summarize().value))
             }
             ValueMode::Filename => {
-                // TODO: return C:\fakepath\<first of selected files> when a file is selected
-                DOMString::from("")
+                let mut path = DOMString::from("");
+                match *self.filelist.borrow() {
+                    Some(ref fl) => match fl.Item(0) {
+                        Some(ref f) => {
+                            path.push_str("C:\\fakepath\\");
```

Comment:
Interestingly, firefox doesn't seem to follow this. But we'll implement the spec and perhaps file a firefox or spec bug later.